### PR TITLE
[CBRD-23071] REGEXP_SUBSTR function

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-23071

**REGEXP_SUBSTR function**
REGEXP_SUBSTR (_source_str_, _regex_pattern_ [, _position_ [, _occurrence_ [, _match_type_]]])

returns searched sub-string by regex pattern. if nothing searched, this function returns NULL.

Optional arguments:
- _position_: the position in src to start the search. If omitted, the default value is 1.
- _occurrence_: the occurrence of the search. If omitted, the default value is 1.
- _match_type_: A string to configure the behavior of matching
  * c: Case sensitive matching.
  * i: Case insensitive matching.

If any of the arguments is NULL, the return value is NULL.

**This PR should be merged after #1671. To review this PR, please refer the first commit of the PR which starts from c54d426.**

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/parser/type_checking.c (1 incorrect indentation(s) found)
```
 19842 
```
#### File: src/query/fetch.c (1 incorrect indentation(s) found)
```
  3976 	    case F_REGEXP_SUBSTR:
```
#### File: src/query/query_executor.c (1 incorrect indentation(s) found)
```
  1538 	    case F_REGEXP_SUBSTR:
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/query/string_opfunc.c [hunk 1]
```diff
@@ -4775,6 +4777,258 @@ db_string_regexp_replace (DB_VALUE * result, DB_VALUE * args[], int const num_ar
...
+      {
+	/* regex execution error */
+	error_status = (error_status == ER_QSTR_BAD_SRC_CODESET) ? NO_ERROR : error_status;
+	goto exit;
+      }
+    // *INDENT-ON*
+
+    if (is_matched)
+      {
+	/* make result */
```
Comment (shparkcubrid) [inline src/query/string_opfunc.c]: if is_matched is false then result is not modified. is it okay?

File: src/query/string_regex.cpp [hunk 2]
```diff
@@ -486,4 +486,95 @@ namespace cubregex
     return error_status;
   }
 #endif
+
+  int substr (std::string &result, bool &is_matched, const cub_regex_object &reg, const std::string &src,
+	      const int position,
+	      const int occurrence, const INTL_CODESET codeset)
```
Comment (shparkcubrid) [inline src/query/string_regex.cpp]: could you correct the indentation

File: src/query/string_opfunc.c (hunk 1 above)
Comment (hgryoo) [reply src/query/string_opfunc.c]: the result is set as null at the beginning of this function.

File: src/query/string_regex.cpp (hunk 2 above)
Comment (hgryoo) [reply src/query/string_regex.cpp]: i fixed it.

**Pull request merged**